### PR TITLE
Converts marine gloves to greyscale

### PR DESCRIPTION
**Pull request by Mark9013100**
## About The Pull Request

Changes the base marine gloves to use the squad overlay system. Removes the squad specific standard and insulated gloves. Marine gloves are now available in the surplus uniform vendors.

Insulated marine gloves are handled by the engineer vendors. The old gear closets had their insulated gloves removed, however they're not used anywhere in normal gameplay so I don't see this causing any issues.

Branch name is a hold over from the original concept.

Massive thanks to Fikou for code assistance.

## Why It's Good For The Game

Consistency, marine gloves now use the same overlay system as the M3 armors and M10 helmets. Insulated marine gloves have also been converted to this system.

## Changelog
:cl: Mark9013100
tweak: Marine gloves have been converted to use the greyscale system.
add: Marine gloves are now available in surplus uniform vendors.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
the generic marine gloves are light grey, look really ugly

**Comment by Rohesie** [comment]
This might require new sprites.

**Comment by Mark9013100** [comment]
> the generic marine gloves are light grey, look really ugly
> same with radio because it uses norma ss13 sprites

The onmob marine gloves are black.

**Comment by Mark9013100** [comment]
Ran into an issue with regular gloves being named insulated, working on it.

File: code/modules/clothing/gloves/marine_gloves.dm [hunk 1]
```diff
@@ -6,18 +6,19 @@
...
+	item_state = "gloves_marine"
 	siemens_coefficient = 0.6
 	permeability_coefficient = 0.05
 	flags_cold_protection = HANDS
 	flags_heat_protection = HANDS
 	min_cold_protection_temperature = GLOVES_MIN_COLD_PROTECTION_TEMPERATURE
 	max_heat_protection_temperature = GLOVES_MAX_HEAT_PROTECTION_TEMPERATURE
 	flags_armor_protection = HANDS
+	var/flags_marine_gloves = GLOVES_SQUAD_OVERLAY
+	var/glove_overlays[]
```
Comment (Rohesie) [inline code/modules/clothing/gloves/marine_gloves.dm]:
```suggestion
	var/list/glove_overlays
```

Comment (Rohesie) [reply code/modules/clothing/gloves/marine_gloves.dm]: Is this variable used anywhere?

Comment (Mark9013100) [reply code/modules/clothing/gloves/marine_gloves.dm]: good question, I'll remove it and check

Comment (Mark9013100) [reply code/modules/clothing/gloves/marine_gloves.dm]: removed the var, encountered no issues

**Comment by Rohesie** [comment]
Is this tested locally?

**Comment by Mark9013100** [comment]
Yes, squad colors worked with all four squads, plus the insulated variant.

**Comment by Rohesie** [comment]
Honestly this is not the best method, but the effort is appreciated. We'll test it.

File: code/datums/jobs/squads.dm [hunk 2]
```diff
@@ -83,6 +84,11 @@ GLOBAL_LIST_EMPTY(helmetmarkings_sl)
 	var/image/armors1 = image('icons/mob/suit_1.dmi',icon_state = "std-armor")
 	armor.color = color
 	armors1.color = color
+	var/image/gloves = image('icons/mob/hands.dmi',icon_state = "std-gloves")
+	var/image/gloves1 = image('icons/mob/hands.dmi',icon_state = "std-gloves")
+	gloves.color = color
+	gloves1.color = color
```
Comment (psykzz) [inline code/datums/jobs/squads.dm]: is `gloves1` needed here?

Comment (Mark9013100) [reply code/datums/jobs/squads.dm]: I'll check.

Comment (Mark9013100) [reply code/datums/jobs/squads.dm]: Removed the var, encountered no issues or runtimes from local testing.

**Comment by Mark9013100** [comment]
oh dear that's a bad travis

File: code/datums/jobs/squads.dm (hunk 2 above)
Comment (Rohesie) [reply code/datums/jobs/squads.dm]: Remove it again? This variable shouldn't be needed.

**Comment by Rohesie** [comment]
`gloves1` is not being used anywhere.

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 3]
```diff
@@ -750,23 +751,18 @@
...
 			if("Charlie")
 				products2 = list(/obj/item/radio/headset/mainship/marine/charlie = 20,
-								/obj/item/clothing/gloves/marine/charlie = 10,
 								/obj/item/clothing/mask/bandanna/charlie = 10)
 			if("Delta")
 				products2 = list(/obj/item/radio/headset/mainship/marine/delta = 20,
-								/obj/item/clothing/gloves/marine/delta = 10,
 								/obj/item/clothing/mask/bandanna/delta = 10)
 	else
-		products2 = list(/obj/item/radio/headset/mainship = 10,
```
Comment (Rohesie) [inline code/game/objects/machinery/vending/marine_vending.dm]: I think you wanted to remove the gloves here, not the headset.

**Pull request merged**